sklearn/metrics/scorer.py

```diff
@@ -126,7 +126,12 @@ def __call__(self, clf, X, y, sample_weight=None):
         y_type = type_of_target(y)
         y_pred = clf.predict_proba(X)
         if y_type == "binary":
-            y_pred = y_pred[:, 1]
+            if y_pred.shape[1] == 2:
+                y_pred = y_pred[:, 1]
+            else:
+                raise ValueError('Must use classifier with two classes'
```

Comment:
should we say "got predict_proba of shape"?

Reply:
That seems like a good idea, because that is what is actually happening.  

*But*, will the *user* understand that?  Is there some place in the documentation where we can write:
>When obtaining predictions from a classifier, if it is only one class, the returned ValueError will be "got predict_proba of shape".  That means it returns one column all with same probabilities (equal to 1). 
**Q:** How to fix this?
**A:** Must use a classifier with 2 classes.

Or will they be able to see it from the test below:  
`with pytest.raises(ValueError, match="use classifier with two classes")`
I would feel better if the user had explicit feedback on how to solve the error.

cc:  @AMDonati

Reply:
Are you saying the current error message is not informative enough? The test is just to make sure that the error that is raised is actually informative.
I suggested adding the predict_proba shape to the error message, not replace the message that's there now.

Reply:
That works.
I tried making the update myself but ran into multiple problems:  [errors list](https://gist.github.com/reshamas/93af459fd61aed10ff3c162ca62a8a98)

Reply:
ok, making progress.  Please let me know next steps. 

cc:  @AMDonati